### PR TITLE
Fixed #26718 -- Added system check for existence of the fields specified by ForeignKey.to_field.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/26718

### CODE REVIEW

**Comment by charettes** [comment]
The formulation `The related model '%s' has no field named '%s'.` sounds better to me. Other than that LGTM.

**Comment by timgraham** [comment]
Please pick a new code as fields.E308/9 are "claimed" in https://github.com/django/django/pull/6640.

File: django/db/models/fields/related.py [hunk 1]
```diff
@@ -463,9 +463,31 @@ def __init__(self, to, on_delete, from_fields, to_fields, rel=None, related_name
...
+
+        errors = []
+        for to_field in self.to_fields:
+            if to_field:
+                try:
+                    self.remote_field.model._meta.get_field(to_field)
+                except exceptions.FieldDoesNotExist:
+                    errors.append(
+                        checks.Error(
+                            "The related model '%s' has no field named '%s'."
```
Comment (timgraham) [inline django/db/models/fields/related.py]: Would this message make it more clear about what argument of the field is problematic?
```
"The 'to_field' '%s' doesn't exist on the related "
"model '%s'."
% (to_field, self.remote_field.model._meta.label),
```

Comment (sir-sigurd) [inline django/db/models/fields/related.py]: It looks more clear, thanks.

File: django/db/models/fields/related.py [hunk 2]
```diff
@@ -463,9 +463,32 @@ def __init__(self, to, on_delete, from_fields, to_fields, rel=None, related_name
 
     def check(self, **kwargs):
         errors = super(ForeignObject, self).check(**kwargs)
+        errors.extend(self._check_to_fields_exist())
         errors.extend(self._check_unique_target())
         return errors
 
+    def _check_to_fields_exist(self):
+        if isinstance(self.remote_field.model, six.string_types):
```
Comment (timgraham) [inline django/db/models/fields/related.py]: Could you add a test that fails if this condition is removed?

File: django/db/models/fields/related.py [hunk 3]
```diff
@@ -463,9 +463,32 @@ def __init__(self, to, on_delete, from_fields, to_fields, rel=None, related_name
...
+                    errors.append(
+                        checks.Error(
+                            "The 'to_field' '%s' doesn't exist on the related "
+                            "model '%s'."
+                            % (to_field, self.remote_field.model._meta.label),
+                            obj=self,
+                            id='fields.E312',
+                        )
+                    )
+
```
Comment (timgraham) [inline django/db/models/fields/related.py]: chop blank line

**Comment by timgraham** [comment]
merged in 21130ce1a9c4fcbfce4c614be9e5408b43092bf0, thanks!

**Pull request closed without merging**